### PR TITLE
Fixing possible memory issues and adding metrics for account prio fees

### PR DESCRIPTION
**Pull request by godmodegalactus**

### CODE REVIEW

File: prioritization_fees/src/account_priofees.rs [hunk 1]
```diff
@@ -16,6 +17,17 @@ use crate::{
     rpc_data::{AccountPrioFeesStats, AccountPrioFeesUpdateMessage},
 };
 
+lazy_static::lazy_static! {
+    static ref WRITE_ONLY_ACCOUNTS_IN_PRIO_CACHE: GenericGauge<prometheus::core::AtomicI64> =
```
Comment (grooviegermanikus) [inline prioritization_fees/src/account_priofees.rs:21]: we should call the "thing" _priofees_ and not _prio_,

File: prioritization_fees/src/account_priofees.rs [hunk 2]
```diff
@@ -178,24 +190,41 @@ impl AccountPrioStore {
...
+            .account_by_prio_fees_all
+            .iter()
+            .map(|v| v.stats_by_slot.len())
+            .sum();
+        let dp_writeonly: usize = self
+            .account_by_prio_fees_writeonly
+            .iter()
+            .map(|v| v.stats_by_slot.len())
+            .sum();
+        NUMBER_OF_PRIO_DATA_POINTS.set((dp_all + dp_writeonly) as i64);
```
Comment (grooviegermanikus) [inline prioritization_fees/src/account_priofees.rs:223]: is that correct? _all_ should include _writeonly_ - no?

File: prioritization_fees/src/account_priofees.rs [hunk 3]
```diff
@@ -178,24 +190,41 @@ impl AccountPrioStore {
         let min_slot_to_retain = produced_block
             .slot
             .saturating_sub(self.number_of_slots_to_save as u64);
-        let cleanup_functor = |mut iter: RefMutMulti<'_, Pubkey, AccountPrio>| {
+        let cleanup_functor = |iter: &mut AccountPrio| {
             while let Some((k, _)) = iter.stats_by_slot.first_key_value() {
```
Comment (grooviegermanikus) [inline prioritization_fees/src/account_priofees.rs:194]: you can use the BtreeMap.iter() which gives the entries sorted by key

**Review by grooviegermanikus — CHANGES_REQUESTED** [review]
smal changes

File: prioritization_fees/src/account_priofees.rs (hunk 2 above)
Comment (godmodegalactus) [reply prioritization_fees/src/account_priofees.rs:223]: No we mangage for both all and writeonly

File: prioritization_fees/src/account_priofees.rs (hunk 3 above)
Comment (godmodegalactus) [reply prioritization_fees/src/account_priofees.rs:194]: Yes that is why i get the first element compare and pop it out. In C++ i know how to iterate and remove multiple first elements. But in rust i cannot find the way.

**Pull request merged**